File: test_api.js
// Import axios
import axios from 'axios';
const BASE_URL = "http://localhost:8000"; // Base URL of your FastAPI backend

// Function to fetch all influencers
async function getInfluencers() {
    try {
        const response = await axios.get(`${BASE_URL}/influencers`);
        console.log("Influencers:", response.data);
    } catch (error) {
        console.error("Error fetching influencers:", error.response ? error.response.status : error.message);
    }
}
// Function to fetch all content
async function getContent() {
    try {
        const response = await axios.get(`${BASE_URL}/content`);
        console.log("Content:", response.data);
    } catch (error) {
        console.error("Error fetching content:", error.response ? error.response.status : error.message);
    }
}
// Function to update a vote (PUT request)
// Function to update vote for an influencer
function updateVote(influencerId, isGoodVote) {
    // Prepare the data for updating the vote
    const voteUpdateData = isGoodVote
        ? { good_vote: 1, bad_vote: 0 }  // Increment good_vote by 1, no change to bad_vote
        : { good_vote: 0, bad_vote: 1 }; // Increment bad_vote by 1, no change to good_vote

    // URL pointing to the FastAPI endpoint
    const url = `http://127.0.0.1:8000/Votes/${influencerId}`;

    // Make a PUT request using fetch API
    fetch(url, {
        method: 'PUT',
        headers: {
            'Content-Type': 'application/json',
        },
        body: JSON.stringify(voteUpdateData),
    })
    .then(response => {
        if (response.ok) {
            return response.json(); // Parse JSON response
        } else if (response.status === 404) {
            throw new Error('Vote not found');
        } else {
            throw new Error('Failed to update vote');
        }
    })
    .then(data => {
        console.log('Vote updated successfully:', data);
        // Optionally, update the UI here based on the response
    })
    .catch(error => {
        console.error('Error:', error.message);
    });
}

// Example usage:
// To increment good vote (thumbs up) for influencer with ID 19
document.getElementById('thumbsUpButton').addEventListener('click', () => {
    updateVote(19, true);  // true indicates thumbs up (good vote)
});

// To increment bad vote (thumbs down) for influencer with ID 19
document.getElementById('thumbsDownButton').addEventListener('click', () => {
    updateVote(19, false);  // false indicates thumbs down (bad vote)
});

File: test_api.py
import requests

# Define the base URL of your FastAPI API
BASE_URL = "http://127.0.0.1:8000"
 # when we use uvicorn to run the API on our local machine it will generate a URL like this then we can use that to make requests to the API

# we can also implement the same in javascript using the fetch API for our frontend
# Function to get all influencers
def get_influencers():
    response = requests.get(f"{BASE_URL}/influencers")
    
    if response.status_code == 200:
        influencers = response.json()
        print("Influencers:")
        for influencer in influencers:
            print(influencer)
    else:
        print(f"Failed to fetch influencers. Status code: {response.status_code}")

# Function to get all content
def get_content():
    response = requests.get(f"{BASE_URL}/content")
    
    if response.status_code == 200:
        content = response.json()
        print("Content:")
        for item in content:
            print(item)
    else:
        print(f"Failed to fetch content. Status code: {response.status_code}")


# Function to update a vote (PUT request)
import requests

def update_vote(influencer_id, is_good_vote):
    """
    Update the vote for a given influencer by incrementing either good_vote or bad_vote by 1.

    Parameters:
    influencer_id (int): The ID of the influencer.
    is_good_vote (bool): If True, increment good votes. If False, increment bad votes.
    """
    
    # Prepare the data for updating the vote
    if is_good_vote:
        vote_update_data = {
            'good_vote': 1,  # Increment good vote by 1
            'bad_vote': 0    # No change to bad votes
        }
    else:
        vote_update_data = {
            'good_vote': 0,  # No change to good votes
            'bad_vote': 1    # Increment bad vote by 1
        }

    # URL pointing to the locally running FastAPI application
    url = f'http://127.0.0.1:8000/Votes/{influencer_id}'

    # Making a PUT request to update the vote
    try:
        response = requests.put(url, json=vote_update_data)
        if response.status_code == 200:
            print('Vote updated successfully:', response.json())
        elif response.status_code == 404:
            print('Vote not found:', response.text)
        else:
            print('Failed to update vote:', response.status_code, response.text)
    except requests.ConnectionError as e:
        print('Failed to connect:', e)



# Update a vote for influencer 18 and content 13 with a thumbs up


# Main function to run the app
if __name__ == "__main__":
    # Fetch all influencers
    get_influencers()
    # Fetch all content
    get_content()
    
    # Example usage:
    # To increment good vote (thumbs up) for influencer with ID 19
    update_vote(19, True)

    # To increment bad vote (thumbs down) for influencer with ID 1
    update_vote(5, False)

   